### PR TITLE
Support namespace qualifiers in classes used in Imperative blocks

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Support namespace qualifiers in classes used in Imperative blocks: https://jira.autodesk.com/browse/QNTM-5647
image
image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @QilongTang @ColinDayOrg 

### FYIs

@smangarole

### CODE REVIEW

File: src/Engine/ProtoCore/Utils/CoreUtils.cs [hunk 1]
```diff
@@ -670,6 +724,38 @@ public static AssociativeNode CreateNodeByCombiningIdentifiers(IList<Associative
             return newIdentList;
         }
 
+        public static AST.ImperativeAST.ImperativeNode CreateNodeByCombiningIdentifiers(IList<AST.ImperativeAST.ImperativeNode> nodeList)
```
Comment (mjkkirschner) [inline src/Engine/ProtoCore/Utils/CoreUtils.cs]: summary tag for public method?

Comment (mjkkirschner) [reply src/Engine/ProtoCore/Utils/CoreUtils.cs]: Any reason not to use an `IEnumerable` for a public method?

Comment (mjkkirschner) [reply src/Engine/ProtoCore/Utils/CoreUtils.cs]: Ah sorry, did not see it was an `IList`

**Comment by ColinDayOrg** [comment]
Generally, LGTM, although it is a little difficult to parse some of the changes. Also, there are some sections of code that were commented out, but left in, was this for any specific reason?

**Pull request merged**